index.bs

```diff
@@ -3484,6 +3641,301 @@ partial dictionary MLOpSupportLimits {
+  </tr>
+  <tr>
+    <td>{{zeroPoint}}</td>
+    <td>{{MLOperandDataType/"uint8"}}, {{MLOperandDataType/"uint32"}}, {{MLOperandDataType/"int32"}}</td>
+    <td>[=/same rank as|same as=] {{input}}</td>
+  </tr>
+  <tr>
+    <td>*output*</td>
+    <td>[=/same type as|same as=] {{zeroPoint}}</td>
+    <td>[=/same rank as|same as=] {{input}}</td>
+  </tr>
+</table>
+
+{{MLOpSupportLimits}} has the following member for {{MLGraphBuilder/quantizeLinear()}}:
+<dl dfn-type=dict-member dfn-for=MLOpSupportLimits>
+    : <dfn>quantizeLinear</dfn>
+    :: Support limits for operator {{MLGraphBuilder/quantizeLinear()}}.
+</dl>
+
+
+<details open algorithm>
+  <summary>
+    The <dfn method for=MLGraphBuilder>quantizeLinear(|input|, |scale|, |zeroPoint|, |options|)</dfn> method steps are:
+  </summary>
+    1. If [=this=].{{MLGraphBuilder/[[hasBuilt]]}} is true, then [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
+    1. If [=MLGraphBuilder/validating operand=] with [=this=] and any of |input|, |scale|, and |zeroPoint| returns false, then [=exception/throw=] a {{TypeError}}.
+    1. If |scale|'s [=MLOperand/rank=] or |zeroPoint|'s [=MLOperand/rank=] mismatches |input|'s [=MLOperand/rank=], then [=exception/throw=] a {{TypeError}}.
+    1. If |scale|'s [=MLOperand/shape=] mismatches |zeroPoint|'s [=MLOperand/shape=], then [=exception/throw=] a {{TypeError}}.
+    1. If [=blockwise broadcasting=] |scale|'s [=MLOperand/shape=] and |input|'s [=MLOperand/shape=] returns failure, then [=exception/throw=] a {{TypeError}}.
+    1. If [=blockwise broadcasting=] |zeroPoints|'s [=MLOperand/shape=] and |input|'s [=MLOperand/shape=] returns failure, then [=exception/throw=] a {{TypeError}}.
+    1. Let |outputDescriptor| be the result of [=creating an MLOperandDescriptor=] given |input|'s [=MLOperand/dataType=] and |input|'s [=MLOperand/shape=].
+    1. *Make graph connections:*
+        1. Let |output| be the result of [=creating an MLOperand=] given [=this=] and |outputDescriptor|.
+        1. Let |operator| be an [=operator=] for the "quantizeLinear" operation, given |input|, |scale|, |zeroPoint|, and |options|.
+        1. Set |output|.{{MLOperand/[[operator]]}} to |operator|.
+        1. Set |operator|'s [=operator/input=] to |input|.
+        1. Set |operator|'s [=operator/output=] to |output|.
+    1. Return |output|.
+</details>
+
+<div class="note">
+  <details open>
+    <summary>
+    The behavior of this operation can be [EMULATED]
+    </summary>
+    This emulation relies on a pending `roundEven` operator in <a href="https://github.com/webmachinelearning/webnn/issues/817">[Issue webnn#817]</a>.
+    <pre highlight="js">
+    function dequantizeLinear(builder, input, scale, zeroPoint, options) {
+      // output = clamp(roundEven(input / scale) + zeroPoint, 0, 255)
+      const floatZeroPoint = builder.cast(zeroPoint, scale.dataType);
+      const upsampledScale = blockwiseBroadcast(builder, scale, input.shape);
```

Comment:
Add a code comment that `blockwiseBroadcast()` is defined in the decomposition for `dequantizeLinear` ?